src/Authentication/AuthenticationManager.php

```diff
@@ -103,7 +104,7 @@ public function getAccount(RequestInterface $request, $cache = TRUE) {
 
     if (!$account) {
 
-      if ($this->plugins->count() && !$this->getIsOptional()) {
+      if (RestfulManager::isRestfulPath($request) && $this->plugins->count() && !$this->getIsOptional()) {
```

Comment:
> how come this was invoked if this wasn't a restful path?

When we are building the menu router we get all the plugins. A part of the plugin initialization is setting the account provided by the `AuthenticationManager`.

> do we care if it's not a restful path?

Yes, otherwise the request object (among others) is not prepared and it breaks.

> I for example already had a cases where I've manually invoked the authentication manager for non restful path.

What was the use case?